ollama_dl.py

```diff
@@ -40,43 +39,82 @@ class DownloadJob:
     blob_url: str
     size: int
 
+async def download_with_resume(client: httpx.AsyncClient, url: str, dest_path: pathlib.Path, total_size: int, progress, task):
+    temp_path = dest_path.with_suffix(f".tmp-{time.time()}")
+    headers = {}
+    mode = "wb"
+    
+    if temp_path.exists():
```

Comment:
Since the temporary path is a new one every time we enter this function, and the current time, fractional seconds and all included, there's practically zero chance that this will ever be true. 🤔